Adaptors/Amqp/src/ConnectionAmqp.cs

```diff
@@ -77,9 +103,46 @@ public async Task<IConnectionAmqp> OpenConnectionAsync()
                                                                   };
     }
 
-    Connection = await connectionFactory.CreateAsync(address)
-                                        .ConfigureAwait(false);
+    var retry = 0;
+    for (; retry < options_.MaxRetries; retry++)
+    {
+      try
+      {
+        Connection = await connectionFactory.CreateAsync(address)
+                                            .ConfigureAwait(false);
+        Connection.AddClosedCallback((x,
+                                      e) => OnCloseConnection(x,
+                                                              e,
+                                                              logger_));
+        break;
+      }
+      catch
+      {
+        logger_.LogInformation("Retrying to create connection");
+        Thread.Sleep(1000 * retry);
```

Comment:
```suggestion
        Task.Delay(TimeSpan.FromSeconds(retry))
```
`Thread.Sleep` is blocking.

Reply:
Needs an await though